### ISSUE TITLE
ERROR: Time out. 

### ISSUE THREAD

**Issue by xiaoqge** [blkchain/blkchain#11]
@grisha  Sorry to interrupt you, want to get some help, thank you very much，Now I don’t know why, when you read 688652, you will find that the data of this block is not read normally,Look at the log and find the error ERROR: Time out. The following is the data detected by my sql, this block has lost a lot of data

```
bitcoinblocks=# select * from txouts where tx_id in (select tx_id from block_txs where block_id =688652);
   tx_id   | n |   value   |                                                  scriptpubkey                                                  | spent 
-----------+---+-----------+----------------------------------------------------------------------------------------------------------------+-------
 651269600 | 0 | 625000000 | \x76a914c825a1ecf2a6830c4401620c3a16f1995057c2ab88ac                                                           | f
 651269600 | 1 |         0 | \x6a34486174685d6021d30fb9ab3e2de65b8c0d8943f1bbb9c8bab086c2d34dd01f4eafa5681c4a77f2c3171148c39dedc141cda8dedc | f
 651269600 | 2 |         0 | \x6a4c2952534b424c4f434b3a0bde938d409648c985a31b79d335172cc03b2dce726b9d9284309d2800347b77                     | f
 651269600 | 3 |         0 | \x6a24b9e11b6d456119451d3227d9af6aca1066eba30454688c5f5235cd326cebba4fa09839bc                                 | f

```

I deleted the data above 688600 and tried to synchronize once, but the problem is still not solved，Are you using it to synchronize your data? Have you encountered this problem?

**Comment by grisha** [comment]
I'm not seeing this problem....

I suspect that one of the prevouts referenced in this block is missing (just a theory), perhaps you will need to roll back further than 688600...

**Comment by xiaoqge** [comment]
thank you

**Comment by xiaoqge** [comment]
@grisha My goodness, I rolled back to 660,000, and I also rolled back the bitcoind node to 660,000 and synced together, but when the block reached 688652, the data began to be missing. After that, everything was normal, and I suspected that the data was not stored in.

**Comment by grisha** [comment]
I have mine running and there is no problem as far as I can tell on my end.

Something happened with your data prior to 660,000 most likely.

BTW - there is no reason to revert the bitcoind, you have to trust it's syncing correctly or we are all doomed :)

Basically what happens is that a block (or part of it) is missing in Postgres, and because of this some transactions will not be written correctly because of referential integrity in postgres, and then other transactions that depend on those will fail, and it just goes on and on.

The fix is just to roll back further. Which is why I recommend putting your postgres data on ZFS and take periodic ZFS snaphots. Whenever this sort of thing happens, you zfs rollback to the last known good state and let it resync from there.

For what it's worth, ever since I pushed that fix 0ae4897 this has not happened and it's been working without any errors at all for me.

**Comment by xiaoqge** [comment]
thank you
